### PR TITLE
OCPBUGS-17417: pull newer busybox from docker until upstream can be updated.

### PR DESCRIPTION
**Pull request by jaypoulz**
#### What type of PR is this?
/kind bug
/kind failing-test
/kind regression

#### What this PR does / why we need it:
This PR is a temporary workaround to update the busybox image used for in our e2e tests. We determined that the current image fails to deploy when running in a privileged context with selinuxuser_execmod=off. We will try to get this accepted upstream by backporting: https://github.com/kubernetes/kubernetes/pull/119636#issuecomment-1678231294

#### Which issue(s) this PR fixes:
Fixes OCPBUGS-17417

#### Does this PR introduce a user-facing change?
```release-note
None
```

### CODE REVIEW

File: test/utils/image/manifest.go [hunk 1]
```diff
@@ -242,7 +242,7 @@ func initImageConfigs(list RegistryList) (map[ImageID]Config, map[ImageID]Config
 	configs[AuthenticatedWindowsNanoServer] = Config{list.GcAuthenticatedRegistry, "windows-nanoserver", "v1"}
 	configs[APIServer] = Config{list.PromoterE2eRegistry, "sample-apiserver", "1.17.7"}
 	configs[AppArmorLoader] = Config{list.PromoterE2eRegistry, "apparmor-loader", "1.4"}
-	configs[BusyBox] = Config{list.PromoterE2eRegistry, "busybox", "1.29-4"}
+	configs[BusyBox] = Config{list.DockerLibraryRegistry, "busybox", "1.36.1"}
```
Comment (soltysh) [inline test/utils/image/manifest.go:245]: I haven't seen that change in upstream, so why do we need it?

**Review by soltysh — CHANGES_REQUESTED** [review]
The commit needs to be named:
`UPSTREAM: 119636: update the busybox test image to 1.36.1` 
to allow us to properly include it in bumps

**Comment by jaypoulz** [comment]
No longer needed. Fixed upstream.

**Pull request closed without merging**